lib/propolis/src/hw/virtio/block.rs

```diff
@@ -192,13 +234,14 @@ impl block::Device for PciVirtioBlock {
 
     fn complete(
         &self,
-        _op: block::Operation,
+        op: block::Operation,
         res: block::Result,
         payload: Box<block::BlockPayload>,
     ) {
-        let mut chain: Box<Chain> =
+        let mut payload: Box<CompletionPayload> =
             payload.downcast().expect("payload must be correct type");
-        self.complete_req(res, &mut chain);
+        let CompletionPayload { rid, ref mut chain } = *payload;
```

Comment:
I'm sad that the stabilization of `Box::into_inner()` seems to have stalled.

Reply:
Yea, I had initially reached for that until I realized 😓